msal/tests/Test.MSAL.NET.Unit/PublicClientApplicationTests.cs

```diff
@@ -1087,5 +1087,87 @@ public void GetUserTest()
             Assert.AreEqual(userToFind.Identifier, fetchedUser.Identifier);
             Assert.AreEqual(userToFind.Name, fetchedUser.Name);
         }
+
+        [TestMethod]
+        [Description("Test for AcquireToken with user canceling authentication")]
+        public async Task AcquireTokenWithAuthenticationCanceledTestAsync()
+        {
+            PublicClientApplication app = new PublicClientApplication(TestConstants.ClientId);
+
+            // Interactive call and user cancels authentication
+            MockWebUI ui = new MockWebUI()
+            {
+                MockResult = new AuthorizationResult(AuthorizationStatus.UserCancel,
+                    TestConstants.AuthorityHomeTenant + "?error=user_canceled")
+            };
+
+            //add mock response for tenant endpoint discovery
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage = MockHelpers.CreateOpenIdConfigurationResponse(TestConstants.AuthorityHomeTenant)
+            });
+
+            MsalMockHelpers.ConfigureMockWebUI(ui);
+
+            try
+            {
+                AuthenticationResult result = await app.AcquireTokenAsync(TestConstants.Scope).ConfigureAwait(false);
+            }
+            catch (MsalClientException exc)
+            {
+                Assert.IsNotNull(exc);
+                Assert.AreEqual("authentication_canceled", exc.ErrorCode);
```

Comment:
>ErrorCode [](start = 63, length = 9)

I would assume this one should be user_canceled ?

Reply:
I changed this. made the ErrorCode `user_cancled_authentication` and removed the message, which had the same thing. lmk if you were thinking of something else.

Reply:
pls. see other comments... probably easier to discuss in person.

---
In reply to: [199969044](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1107#discussion_r199969044) [](ancestors = 199969044)